src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -535,6 +535,24 @@ def _get_components_pairs(
 
         return bat_inv_map, inv_bat_map
 
+    def _get_working_batteries(self, batteries: Set[int]) -> Set[int]:
+        """Get subset with working batteries.
+
+        If there are no working batteries in the given set,
+        then return input set to use them all in case of gap discharge.
```

Comment:
I think you should say in the PR description that this is being done for gap discharge,  and keep it generic in the docstring,
```suggestion
        If none of the given batteries are working, then treat all of them
        as working.
```

Reply:
Done